crypto/evp/p_lib.c

```diff
@@ -237,6 +237,17 @@ void *EVP_PKEY_get0(const EVP_PKEY *pkey)
     return pkey->pkey.ptr;
 }
 
+const unsigned char *EVP_PKEY_get_hmac(const EVP_PKEY *pkey, int *len)
```

Comment:
Should be `EVP_PKEY_get0_hmac` or perhaps better yet `EVP_PKEY_get0_hmac_key`?

size_t for the length.

The type check should happen before you call EVP_PKEY_get0.